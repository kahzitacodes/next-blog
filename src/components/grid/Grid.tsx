import { PropType } from './Grid.types'

export const Grid = ({
  children,
  cols = 1,
  gap = 8,
  xs,
  sm,
  md,
  lg,
  xl,
  xxl
}: PropType) => {
  const extraSmall = xs ? `xs:grid-cols-${xs.cols} xs:gap-${xs.gap}` : ''
  const small = sm ? `sm:grid-cols-${sm.cols} sm:gap-${sm.gap}` : ''
  const medium = md ? `md:grid-cols-${md.cols} md:gap-${md.gap}` : ''
  const large = lg ? `lg:grid-cols-${lg.cols} lg:gap-${lg.gap}` : ''
  const extraLarge = xl ? `xl:grid-cols-${xl.cols} xl:gap-${xl.gap}` : ''
  const extraExtraLarge = xxl
    ? `2xl:grid-cols-${xxl.cols} 2xl:gap-${xxl.gap}`
    : ''

  const responsive = `${extraSmall} ${small} ${medium} ${large} ${extraLarge} ${extraExtraLarge}`

  return (
    <div className={`grid w-full grid-cols-${cols} gap-${gap} ${responsive}`}>
      {children}
    </div>
  )
}
