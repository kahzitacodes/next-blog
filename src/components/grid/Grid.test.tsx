import { render, screen } from '@/utils/tests/custom-render'
import { Grid } from './Grid'

describe('<Grid />', () => {
  it('should render successfully', () => {
    const { container } = render(
      <Grid dataTestid="grid" sm={{ cols: 2, gap: 6 }} lg={{ cols: 3, gap: 8 }}>
        {Array.from({ length: 9 }, (_, index: number) => (
          <div key={index}>grid</div>
        ))}
      </Grid>
    )

    expect(screen.getByTestId('grid')).toHaveClass(
      'lg:grid-cols-3',
      'sm:grid-cols-2'
    )
    expect(container).toMatchSnapshot()
  })
})
