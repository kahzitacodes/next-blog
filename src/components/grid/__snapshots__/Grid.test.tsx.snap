// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> should render successfully 1`] = `
<div>
  <div
    class="grid w-full grid-cols-1 gap-8  sm:grid-cols-2 sm:gap-6  lg:grid-cols-3 lg:gap-8  "
    data-testid="grid"
  >
    <div>
      grid
    </div>
    <div>
      grid
    </div>
    <div>
      grid
    </div>
    <div>
      grid
    </div>
    <div>
      grid
    </div>
    <div>
      grid
    </div>
    <div>
      grid
    </div>
    <div>
      grid
    </div>
    <div>
      grid
    </div>
  </div>
</div>
`;
